### ISSUE TITLE
--reprint-on-exit/-X no longer functions

### ISSUE THREAD

**Issue by dmfay** [okbob/pspg#180]
I upgraded from 5.0.5 to 5.1.1, and I'm using it from `psql` with the following config:

```
\setenv PSQL_PAGER 'pspg --ignore-case --no-sound --quit-if-one-screen --only-for-tables --reprint-on-exit -s 22'
```

When I quit (`q`) after inspecting large result sets in pspg, the pager view is no longer reprinted before the next prompt:

```
[local] dian#dian= \df+
<pspg opens and closes completely>
[local] dian#dian=
```

**Comment by okbob** [comment]
Hi

út 20. 7. 2021 v 20:15 odesílatel Dian Fay ***@***.***>
napsal:

> I upgraded from 5.0.5 to 5.1.1, and I'm using it from psql with the
> following config:
>
> \setenv PSQL_PAGER 'pspg --ignore-case --no-sound --quit-if-one-screen --only-for-tables --reprint-on-exit -s 22'
>
> When I quit (q) after inspecting large result sets in pspg, the pager
> view is no longer reprinted before the next prompt:
>
> [local] dian#dian= \df+
> <pspg opens and closes completely>
> [local] dian#dian=
>
>
can me send an log, please?

use the "log" option

**Comment by dmfay** [comment]
```
Tue Jul 20 14:28:27 2021 [68628] started
Tue Jul 20 14:28:27 2021 [68628] with utf8 support
Tue Jul 20 14:28:27 2021 [68628] sleep 10ms per 1000 rows
Tue Jul 20 14:28:27 2021 [68628] sleep 10ms per 1000 rows
Tue Jul 20 14:28:27 2021 [68628] read rows 2600
Tue Jul 20 14:28:27 2021 [68628] read input 2600 rows
Tue Jul 20 14:28:27 2021 [68628] terminal size by TIOCGWINSZ rows: 57, cols: 93
Tue Jul 20 14:28:27 2021 [68628] ncurses started
Tue Jul 20 14:28:27 2021 [68628] kmous=\E[<, TERM=xterm-256color
Tue Jul 20 14:28:27 2021 [68628] xterm mouse mode 1002 will be used
Tue Jul 20 14:28:27 2021 [68628] xterm mouse mode 1002 activated
Tue Jul 20 14:28:27 2021 [68628] screen size - maxy: 57, maxx: 93
Tue Jul 20 14:28:28 2021 [68628] process command: Quit
Tue Jul 20 14:28:28 2021 [68628] releasing cmd bar before end
Tue Jul 20 14:28:28 2021 [68628] xterm mouse mode 1002 is deactivated
Tue Jul 20 14:28:28 2021 [68628] ncurses ended
Tue Jul 20 14:28:28 2021 [68628] correct quit
```

**Comment by okbob** [comment]
út 20. 7. 2021 v 20:23 odesílatel Pavel Stehule ***@***.***>
napsal:

> Hi
>
> út 20. 7. 2021 v 20:15 odesílatel Dian Fay ***@***.***>
> napsal:
>
>> I upgraded from 5.0.5 to 5.1.1, and I'm using it from psql with the
>> following config:
>>
>> \setenv PSQL_PAGER 'pspg --ignore-case --no-sound --quit-if-one-screen --only-for-tables --reprint-on-exit -s 22'
>>
>> When I quit (q) after inspecting large result sets in pspg, the pager
>> view is no longer reprinted before the next prompt:
>>
>> [local] dian#dian= \df+
>> <pspg opens and closes completely>
>> [local] dian#dian=
>>
>>
> can me send an log, please?
>
> use the "log" option
>

I can repeat it. I'll look at this issue.

Pavel

**Comment by okbob** [comment]
út 20. 7. 2021 v 20:29 odesílatel Dian Fay ***@***.***>
napsal:

> Tue Jul 20 14:28:27 2021 [68628] started
> Tue Jul 20 14:28:27 2021 [68628] with utf8 support
> Tue Jul 20 14:28:27 2021 [68628] sleep 10ms per 1000 rows
> Tue Jul 20 14:28:27 2021 [68628] sleep 10ms per 1000 rows
> Tue Jul 20 14:28:27 2021 [68628] read rows 2600
> Tue Jul 20 14:28:27 2021 [68628] read input 2600 rows
> Tue Jul 20 14:28:27 2021 [68628] terminal size by TIOCGWINSZ rows: 57, cols: 93
> Tue Jul 20 14:28:27 2021 [68628] ncurses started
> Tue Jul 20 14:28:27 2021 [68628] kmous=\E[<, TERM=xterm-256color
> Tue Jul 20 14:28:27 2021 [68628] xterm mouse mode 1002 will be used
> Tue Jul 20 14:28:27 2021 [68628] xterm mouse mode 1002 activated
> Tue Jul 20 14:28:27 2021 [68628] screen size - maxy: 57, maxx: 93
> Tue Jul 20 14:28:28 2021 [68628] process command: Quit
> Tue Jul 20 14:28:28 2021 [68628] releasing cmd bar before end
> Tue Jul 20 14:28:28 2021 [68628] xterm mouse mode 1002 is deactivated
> Tue Jul 20 14:28:28 2021 [68628] ncurses ended
> Tue Jul 20 14:28:28 2021 [68628] correct quit
>
>
please, can you check master branch? Unfortunately, this feature was
broken by refactoring. It should be fixed.

Pavel

**Comment by okbob** [comment]
út 20. 7. 2021 v 21:08 odesílatel Dian Fay ***@***.***>
napsal:

> It's working now, thanks!
>

Thank you very much for report

Pavel

**Comment by okbob** [comment]
it is fixed in 5.1.2
